BMM.Core/Implementations/PlayObserver/PersistedEventWriter.cs

```diff
@@ -21,35 +23,44 @@ public class PersistedEventWriter : IDelayedStartupTask
         private readonly IBlobCache _localStorage;
         private readonly ILogger _logger;
         private readonly IAnalytics _analytics;
+        private readonly IPlaybackHistoryService _playbackHistoryService;
 
-        public PersistedEventWriter(IPlayStatistics playStatistics, IBlobCache localStorage, ILogger logger, IAnalytics analytics)
+        public PersistedEventWriter(
+            IPlayStatistics playStatistics,
+            IBlobCache localStorage,
+            ILogger logger,
+            IAnalytics analytics,
+            IPlaybackHistoryService playbackHistoryService)
         {
             _playStatistics = playStatistics;
             _localStorage = localStorage;
             _logger = logger;
             _analytics = analytics;
+            _playbackHistoryService = playbackHistoryService;
         }
 
         public async Task RunAfterStartup()
         {
             var storedEvent = await _localStorage.GetOrCreateObject<TrackPlayedEvent>(StorageKeys.UnfinishedTrackPlayedEvent, () => null);
+
             if (storedEvent == null)
-            {
                 return;
-            }
 
             storedEvent.SentAfterStartup = true;
 
             _logger.Info("PersistedEventLogger", "Logging Track played after startup");
             _analytics.LogEvent("Log Track played after startup",
                 new Dictionary<string, object>
                 {
-                    {"TrackId", storedEvent.TrackId},
-                    {"TimestampStart", storedEvent.TimestampStart},
-                    {"TimestampEnd", storedEvent.TimestampEnd},
-                    {"SpentTime", storedEvent.SpentTime},
-                    {"UniqueSecondsListened", storedEvent.UniqueSecondsListened}
+                    {nameof(storedEvent.TrackId), storedEvent.TrackId},
```

Comment:
I thought about using nameof in this case before. But we have several analytics charts that rely on these names. Therefore I decided not to do it, so that if we rename the property, we still get the correct string to the backend without breaking those charts.

Reply:
Ok, reverted